custom/conf/app.example.ini

```diff
@@ -2248,6 +2248,18 @@ PATH =
 ;;
 ;; Enable/Disable user statistics for nodeinfo if federation is enabled
 ; SHARE_USER_STATISTICS = true
+;;
+;; HTTP signature algorithms
+; ALGORITHMS = rsa-sha256, rsa-sha512
```

Comment:
are more than rsa supported?

Reply:
If we only cared about federating between Gitea instances, we could choose other algorithms, but some other AP software like Mastodon only support RSA so we should just stick to `rsa-sha256` for interoperability.

Reply:
does it would make a change/migration to a new one afterwards hard?

yes -> use hs2019 by default and get it upstream in other instances like mastodon
no -> take track and get it upstream first in other instances but support it here too

Reply:
The go-ap/httpsig package supports Ed25519 which is what I expose for my existing ActivityPub services. However I'm not yet worried about federating with Mastodon.

Reply:
> does it would make a change/migration to a new one afterwards hard?
> 
> yes -> use hs2019 by default and get it upstream in other instances like mastodon no -> take track and get it upstream first in other instances but support it here too

No, switching from rsa-sha256 to hs2019 or any other algorithm in the future should be as simple as changing the default value of that `ALGORITHMS` setting.